Examples/ArcGISToolkitExamples/TimeSliderExample.swift

```diff
@@ -0,0 +1,89 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGISToolkit
+import ArcGIS
+
+class TimeSliderExample: MapViewController {
+    
+    private var map : AGSMap?
+    private var timeSlider : TimeSlider?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Create an instance of a map with ESRI topographic basemap
+        self.map = AGSMap(basemap: AGSBasemap.topographic())
+        self.mapView.map = self.map
+        
+        // Configure time slider
+        let timeSlider = TimeSlider()
+        timeSlider.isHidden = true
+        timeSlider.addTarget(self, action: #selector(TimeSliderExample.timeSliderValueChanged(timeSlider:)), for: .valueChanged)
+        view.addSubview(timeSlider)
+            
+        // Add constraints to position the slider
+        let margin: CGFloat = 10.0
+        timeSlider.translatesAutoresizingMaskIntoConstraints = false
+        timeSlider.bottomAnchor.constraint(equalTo: mapView.attributionTopAnchor, constant: -margin).isActive = true
+        timeSlider.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: margin).isActive = true
+        timeSlider.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -margin).isActive = true
+        
+        // Set time slider 
+        self.timeSlider = timeSlider
+        
+        // Add layer
+        let mapImageLayer = AGSArcGISMapImageLayer(url: URL(string: "https://rtc-100-3.esri.com/arcgis/rest/services/timeSupport/cycle_routes/MapServer")!)
+        mapView.map?.operationalLayers.add(mapImageLayer)
+        mapImageLayer.load(completion: { [weak self] (error) in
+            guard error == nil else {
+                print("error: \(String(describing: error))")
+                return
+            }
+            
+            // Zoom to full extent of layer
+            self?.mapView.setViewpoint(AGSViewpoint(targetExtent: mapImageLayer.fullExtent!), completion: nil)
+            
+            self?.timeSlider?.initializeTimeProperties(timeAwareLayer: mapImageLayer, completion: { [weak self] (error) in
+                guard error == nil else {
+                    print("error: \(String(describing: error))")
+                    self?.showError(error!)
+                    return
+                }
+                print("Time slider is successfully initialized!")
+                defer {
+                    self?.timeSlider?.isHidden = false
+                }
+            })
+        })
+    }
+    
+    @objc func timeSliderValueChanged(timeSlider: TimeSlider) {
+        print("Time slider value changed: (\(String(describing: timeSlider.currentExtent?.startTime)), \(String(describing: timeSlider.currentExtent?.endTime)))")
+        if mapView.timeExtent != timeSlider.currentExtent {
```

Comment:
I had the same question regarding `currentExtent` vs `currentTimeExtent` but I went as per design. Since, it is a TimeSlider, it does not make sense to use `Time` in all properties. So I did not raised that question.